tests/php/Authenticator/LoginHandlerTest.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace SilverStripe\MFA\Tests\Authenticator;
+
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\FunctionalTest;
+use SilverStripe\MFA\Authenticator\MemberAuthenticator;
+use SilverStripe\MFA\Method\MethodInterface;
+use SilverStripe\MFA\Model\RegisteredMethod;
+use SilverStripe\MFA\Service\MethodRegistry;
+use SilverStripe\MFA\Store\SessionStore;
+use SilverStripe\MFA\Tests\Stub\BasicMath\Method;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+
+class LoginHandlerTest extends FunctionalTest
+{
+    protected static $fixture_file = 'LoginHandlerTest.yml';
+
+    protected function setUp()
+    {
+        parent::setUp();
+        Config::modify()->set(MethodRegistry::class, 'methods', [Method::class]);
+
+        Injector::inst()->load([
+            'SilverStripe\Security\Security' => [
+                'properties' => [
+                    'authenticators' => [
+                        'default' => '%$' . MemberAuthenticator::class,
+                    ]
+                ]
+            ]
+        ]);
+    }
+
+    public function testMFAStepIsAdded()
+    {
+        $member = $this->objFromFixture(Member::class, 'guy');
+
+        $this->autoFollowRedirection = false;
+        $response = $this->doLogin($member, 'Password123');
+        $this->autoFollowRedirection = true;
+
+        $this->assertSame(302, $response->getStatusCode());
+        $this->assertSame('http://localhost/Security/login/default/mfa', $response->getHeader('location'));
```

Comment:
Where is `http://localhost` mocked? I think maybe you should check for `assertContains('Security/login/default/mfa')` instead to be less rigid. This kind of assertion is likely to fail in combination with fluent or subsites

Reply:
Good point. I switched to `stringEndsWith` but I can quickly update that to `contains` if you're still concerned.